### PR TITLE
Changes to prepare for fake tensors on in functorch by default

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #84447
* #84671
* __->__ #84432

Fixes some errors you run into in dynamo when turning on fake tensors. I'm waiting on flipping the switch because I need to also get some fixes into dynamo + do benchmarking.

I could manually turn off fake tensors in functorch in dynamo, and then turn it on here if requested, although the changes here are pretty minimal.

### CODE REVIEW

File: torch/fx/passes/backends/cudagraphs.py [hunk 1]
```diff
@@ -21,15 +22,17 @@ def is_node_supported(self, submodules, node: torch.fx.Node) -> bool:
 
         found_not_cuda = False
 
+        FK = "val" if functorch.compile.config.use_fake_tensor else "fake_result"
```
Comment (Chillee) [inline torch/fx/passes/backends/cudagraphs.py:25]: I think this should always just be "val" now.

File: torch/_subclasses/fake_tensor.py [hunk 2]
```diff
@@ -843,9 +843,7 @@ def wrap(e, device=None):
             return tree_map(partial(wrap), r)
 
     def may_turn_const(self, t):
-        return (
-            self.const_tensors and t.numel() <= CONSTANT_NUMEL_LIMIT and not t.is_sparse
-        )
+        return self.const_tensors and t.numel() <= CONSTANT_NUMEL_LIMIT and not t.is_sparse and not isinstance(t, FakeTensor)
```
Comment (Chillee) [inline torch/_subclasses/fake_tensor.py:846]: Why is this?

**Review by Chillee — APPROVED** [review]
LGTM

File: torch/fx/passes/backends/cudagraphs.py (hunk 1 above)
Comment (anijain2305) [reply torch/fx/passes/backends/cudagraphs.py:25]: This will also fix the `aot_cudagraphs` 0% passrate observed in the nightlies.

Comment (Chillee) [reply torch/fx/passes/backends/cudagraphs.py:25]: We should just fix all the uses of `fake_result` to use `val` instead.

File: torch/_subclasses/fake_tensor.py (hunk 2 above)
Comment (eellison) [reply torch/_subclasses/fake_tensor.py:846]: you don't want to make a constant when you run an fx graph, and the input is a fake tensor (obviously that's not a constant)

File: torch/fx/passes/backends/cudagraphs.py (hunk 1 above)
Comment (eellison) [reply torch/fx/passes/backends/cudagraphs.py:25]: If we have `aot_cudagraphs` at 0% in nightlies we should temporarily disable the tests

Comment (eellison) [reply torch/fx/passes/backends/cudagraphs.py:25]: im just going to do a polyfill for now and we can update later

**Pull request closed without merging**